omniscidb/ResultSet/ArrowResultSetConverter.cpp

```diff
@@ -1057,29 +1057,17 @@ size_t convert_rowwise(
         int32_t arr_size = 0;
         if (*array_value) {
           arr_size = static_cast<int32_t>((*array_value)->size());
-          // Fixed size array NULL value can be expressed as a zero-length vector.
-          // We should add a proper number of empty values for such case.
-          if (column.col_type->isFixedLenArray() && arr_size != num_elems) {
-            CHECK_EQ(arr_size, 0);
-            append_empty_values_and_validity(elem_type,
-                                             num_elems,
+          for (auto& elem_value : **array_value) {
```

Comment:
Can this dereference fail?

Reply:
No, this is a dereference of `boost::optional` with the preceding check that it is not empty.